pkg/cluster/internal/kubeadm/config.go

```diff
@@ -464,6 +520,13 @@ func Config(data ConfigData) (config string, err error) {
 	// assume the latest API version, then fallback if the k8s version is too low
 	templateSource := ConfigTemplateBetaV2
 	if ver.LessThan(version.MustParseSemantic("v1.12.0")) {
+		// in Kubernetes 1.11 we need to enable this feature, unless the user
+		// explicitly disabled it. this is considered part of our "base config"
+		// so it's totally not a layering violation, we just don't want this logic
+		// encoded in the text template ;-)
+		if _, set := data.FeatureGates["DynamicProvisioningScheduling"]; !set {
+			data.FeatureGates["DynamicProvisioningScheduling"] = true
```

Comment:
this means the kind option will always be set on old versions and will stomp manual feature gate patches, right? probably not a big deal for older versions but something to keep in mind for future internal auto feature gate enablement

Reply:
I think antonio was using the _kubeadm_ feature gate flags which is an additional related way to enable only at most a few specific features. 

A user supplied patch to component feature gates should still be applied on top of this, stomping this value.
Kubeadm feature gates on the other hand that happen to cause defaulting Kubernetes component feature gates are stomped by user supplied feature gate values (including when kind does this).



We do:
- generate base kubeadm config in kind internally, based on kind's cluster config, including this new feature gates field
- take user kubeadm config patches and apply them on top, trumping any genrated config

In general relying on patching kubeadm config is not the most stable but does provide a way to apply overrides to what kind is doing, which is not the most stable plan long term. 

We should highlight that better.

Reply:
though we won't stop the kubeadm feature gates for one particular case if we don't set an explicit emptystring :+)